tsconfig.test.json

```diff
@@ -0,0 +1,18 @@
+{
+  "COMMENT": "This file exists because typescript-eslint/parser needs every file to be explicitly included in a project.",
+  "compilerOptions": {
+    "outDir": "./build",
+    "sourceMap": true,
+    "noImplicitAny": true,
+    "noUnusedLocals": true,
+    "downlevelIteration": true,
+    "experimentalDecorators": true,
+    "module": "commonjs",
+    "target": "es5",
+    "lib": ["es6", "dom"],
+    "declaration": true
+  },
+  "include": [
+    "./test/**/*",
```

Comment:
Why cant we include this in tsconfig.json? The current one include the src and guides folder:
```
  "include": [
    "./src/**/*",
    "./guides/docs.ts"
  ]
```

Reply:
Including files here means they are included in the transpiled output.

The right fix for this is to use project references:

https://www.typescriptlang.org/docs/handbook/project-references.html

to make a test project that depends on the app project. We can do this in a follow-up. It's not a big deal for now, because the second definition is only used for the lint config.

Reply:
Updated commit with project references pushed.